pandas/tests/io/formats/test_format.py

```diff
@@ -1868,6 +1868,33 @@ def test_repr_html(self, float_frame):
 
         tm.reset_display_options()
 
+    def test_repr_html_series(self):
+        data = [1, "two", 3.1, -4.2, True, np.nan]
+
+        fmt.set_option("display.max_rows", len(data))
+
+        small = pd.Series(data, name="test series")
+        sm_html = small._repr_html_()
+
+        for val in data:
+            if pd.isnull(val):
+                assert "<td>NaN</td>" in sm_html
+            else:
+                assert "<td>{}</td>".format(val) in sm_html
+
+        assert "<p>Name: <b>{}</b>".format(small.name) in sm_html
```

Comment:
Can you construct sm_html literally instead? I think small enough to do; would make for stronger assertions

Reply:
I've pushed a commit for this now - personally I still prefer the old approach which is just checking for the meaningful content and is relaxed about other things like HTML indentation which don't affect the meaning of the output. I also think it makes the tests more readable. But let me know which you prefer and I can always revert if you want.